### PR TITLE
fix(progress)!: convert value range to 0-100

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #7207

## Summary
 
Fixes the discrepancy in value range used between Progress (0 -> 1.0) and Loader (0 -> 100), to have components use a similar expected value.

BREAKING CHANGE: Refactors `progress` to use the value range of 0-100. Developers will need to provide a function to map existing values or use the updated range.

### CODE REVIEW

**Review by aPreciado88 — APPROVED** [review]
🚀

**Review by benelan — CHANGES_REQUESTED** [review]
The value being between 0 and 1 comes from the native progress:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress#value

Plus this would be a breaking change in my opinion. If we want consistency, maybe we should change loader instead, which doesn't have a native equivalent.

**Comment by Elijbet** [comment]
> The value being between 0 and 1 comes from the native progress: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress#value
> 
> Plus this would be a breaking change in my opinion. If we want consistency, maybe we should change loader instead, which doesn't have a native equivalent.

We decided not to go with 0-1 because of this little study: https://github.com/Esri/calcite-design-system/issues/7207#issuecomment-2460985536

**Review by jcfranco — CHANGES_REQUESTED** [review]
Please roll back changes unrelated to the progress component.

File: packages/calcite-components/src/components/progress/progress.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class Progress extends LitElement {
    */
   @property({ reflect: true }) type: "indeterminate" | "determinate" = "determinate";
 
-  /** When `type` is `"determinate"`, the component's progress value with a range of 0.0 - 1.0. */
+  /** The component's value. Valid only for `"determinate"` indicators. Percent complete of 100. */
```
Comment (jcfranco) [inline packages/calcite-components/src/components/progress/progress.tsx:37]: +@geospatialem @DitwanP for editorial perusal.

File: packages/calcite-components/src/components/progress/progress.tsx [hunk 2]
```diff
@@ -43,14 +43,14 @@ export class Progress extends LitElement {
...
-    const barStyles = isDeterminate ? { width: `${this.value * 100}%` } : {};
+    const barStyles = isDeterminate ? { width: `${this.value}%` } : {};
     const dir = getElementDir(this.el);
     return (
       <div
         ariaLabel={this.label || this.text}
-        ariaValueMax={1}
-        ariaValueMin={0}
-        ariaValueNow={this.value}
+        ariaValueMax={isDeterminate ? "100" : undefined}
```
Comment (jcfranco) [inline packages/calcite-components/src/components/progress/progress.tsx:51]: Noice!

**Review by jcfranco — APPROVED** [review]
✨🚀✨

File: packages/calcite-components/src/components/progress/progress.tsx (hunk 1 above)
Comment (geospatialem) [reply packages/calcite-components/src/components/progress/progress.tsx:37]: WDYT about:

```suggestion
  /** When `type` is `"determinate"`, specifies the component's value with a range of 0 to 100. */
```

**Review by benelan — APPROVED** [review]
:+1:

**Pull request merged**